### PR TITLE
fix: do not remove registries when podman extension is stopping

### LINKED ISSUE #3135: Configured suggested registry is removed on exit from podman-desktop
### Bug description

After multiple starts and stops .conf/container/auth.json entries being removed. The steps to replicate are below

https://github.com/containers/podman-desktop/assets/620330/b75c942a-b623-476c-b0a7-6de1cd19d28e

### Operating system

macOS M1 13.3

### Installation Method

Installer from website/GitHub releases

### Version

1.1.0

### Steps to reproduce

1. Start podman-desktop
2. Go to Settings/Registries
3. Configure docker registry using your login and password
4. Exit from podman-desktop using tray menu item `Quit`
5. Start podman-desktop
6. Go to Settings/Registries and docker registry entry is still there
7. Exit from podman-desktop using tray menu item `Quit`
8. Start podman-desktop again
9. Go to Settings/Registries and docker registry entry should not be configured

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
do not remove registries when podman extension is stopping

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

fixes #3135 

### How to test this PR?

Use the issue test case

### CODE REVIEW

File: extensions/podman/src/registry-setup.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class RegistrySetup {
     return path.resolve(podmanConfigContainersPath, 'auth.json');
   }
 
-  protected async updateRegistries(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  protected async updateRegistries(_extensionContext: extensionApi.ExtensionContext): Promise<void> {
```
Comment (jeffmaury) [inline extensions/podman/src/registry-setup.ts]: What is the rationale for this ?

Comment (benoitf) [reply extensions/podman/src/registry-setup.ts]: either
1. I remove extensionContext attribute all over the calls
1. or as it's currently not used I make it optional (we could need it when using the context)

I choose option 1

Comment (benoitf) [reply extensions/podman/src/registry-setup.ts]: @jeffmaury you prefer option 1 ?

Comment (benoitf) [reply extensions/podman/src/registry-setup.ts]: @jeffmaury updated by removing unused extensionContext

Comment (benoitf) [reply extensions/podman/src/registry-setup.ts]: @jeffmaury I've proceed with option 1

**Comment by benoitf** [comment]
Applying auto-merge

File: extensions/podman/src/registry-setup.ts [hunk 2]
```diff
@@ -77,7 +77,8 @@ export class RegistrySetup {
     // For each registry in the file that is not in the inMemory, add it
     const toBeAdded = inFileRegistries.filter(fileRegistry => !this.localRegistries.has(fileRegistry.serverUrl));
     toBeAdded.forEach(registry => {
-      extensionContext.subscriptions.push(extensionApi.registry.registerRegistry(registry));
```
Comment (dgolovin) [inline extensions/podman/src/registry-setup.ts:80]: That is how I temporary fixed it in my local codebase.

**Review by dgolovin — APPROVED** [review]

**Pull request merged**